nflow-examples/spring-boot/full-stack-kotlin/src/workflow/ExampleWorkflow.kt

```diff
@@ -0,0 +1,45 @@
+package nflow.kotlin.workflow
+
+import io.nflow.engine.workflow.definition.NextAction
+import io.nflow.engine.workflow.definition.NextAction.moveToStateAfter
+import io.nflow.engine.workflow.definition.StateExecution
+import io.nflow.engine.workflow.definition.WorkflowState
+import io.nflow.engine.workflow.definition.WorkflowStateType
+import io.nflow.engine.workflow.definition.WorkflowStateType.manual
+import io.nflow.engine.workflow.definition.WorkflowStateType.start
+import nflow.kotlin.engine.KtWorkflowDefinition
+import nflow.kotlin.engine.getVar
+import nflow.kotlin.workflow.State.Companion.error
+import nflow.kotlin.workflow.State.Companion.repeat
+import org.joda.time.DateTime.now
+import org.springframework.stereotype.Component
+
+class State(private val name: String, private val type: WorkflowStateType, private val description: String) : WorkflowState {
+    companion object {
+        val repeat = State("repeat", start, "Repeating state")
+        val error = State("error", manual, "Error state")
+    }
+
+    override fun getDescription() = description
+    override fun getType() = type
+    override fun name() = name
+}
+
+@Component
+class ExampleWorkflow : KtWorkflowDefinition<State>(TYPE, repeat, error) {
+
+    init {
+        permit(repeat, repeat)
+    }
+
+    fun repeat(execution: StateExecution): NextAction {
+        println("Counter: " + execution.getVariable(VAR_COUNTER))
+        execution.setVariable(VAR_COUNTER, execution.getVar<Int>(VAR_COUNTER) + 1)
```

Comment:
Where is this `getVar<Int>` method? Is it some kind of kotlin magic thing?

Reply:
Found it.

Reply:
Yes, it's Kotlin extension function, and allows nice inline reified typing of the return value (as a type parameter to the function), instead of requiring to pass the type class as a parameter to the function. Kotlin magic...